edgedb/options.py

```diff
@@ -137,35 +154,54 @@ def with_transaction_options(self, options: TransactionOptions = None):
 
     def with_retry_options(self, options: RetryOptions=None):
         """Returns object with adjusted options for future retrying
-        transactions.
+        transactions and queries outside transactions.
 
         :param options RetryOptions:
             Object that encapsulates retry options.
 
         This method returns a "shallow copy" of the current object
-        with modified transaction options.
+        with modified retry options.
 
         Both ``self`` and returned object can be used after, but when using
-        them transaction options applied will be different.
+        them retry options applied will be different.
         """
 
         result = self._shallow_clone()
         result._options = self._options.with_retry_options(options)
         return result
 
+    def with_timeout_options(self, options: TimeoutOptions = None):
+        """Returns object with adjusted options for future timeouts.
```

Comment:
@tailhook @elprans do we want to use TimeoutOptions here instead of kwargs?

Reply:
The api feels unnecessarily verbose to me.

Reply:
We have discussed this in edgedb/edgedb#3352 and you seemingly agreed with the mixin approach.

Reply:
Hm, this is my comment: https://github.com/edgedb/edgedb/discussions/3352#discussioncomment-2036168, that specifically suggests the `client.with_timeout(command_timeout=5)` api

Reply:
Ah, OK. I got confused.

We need to think how timeouts integrate with retries and if there is a benefit to a pluggable timeout policy like we have with retries.

Reply:
I think the major issue with retries is that they are more complex that one or two numbers: this is a set of rules and we had to have both, the way to add a single rule and the way to replace all rules reliably.

This is not the case for timeouts. So `with_timeout` or similar with kwargs is fine.

We can also have `with_options(command_timeout=5)`, to get rid of repeating `timeout`. However, we may want to keep `with_options` for something that is a combination of retry+transactions+timeout options, if we ever need one.